### PR TITLE
(PCP-766) Restrict permissions on spool files, logfile

### PR DESCRIPTION
**Pull request by MikaelSmith**
Applies permission 0640 to ensure only owner/group can read and only
owner can write sensitive spool files.

Ensure logfile is not world-readable. It may contain sensitive
information in the future.

Blocked on merge and release of https://github.com/puppetlabs/leatherman/pull/249.

### CODE REVIEW

File: lib/src/results_storage.cc [hunk 1]
```diff
@@ -15,6 +16,8 @@
 
 #include <algorithm>  // std::find
 
+const uint16_t UMASK_FLAGS { 026 };
```
Comment (nicklewis) [inline lib/src/results_storage.cc]: This should probably be `0026` for clarity.

**Comment by mruzicka** [comment]
There is only a linting problem which we'll fix in a maint PR.

**Comment by MikaelSmith** [comment]
Hopefully this passes CI. Leatherman 1.1.0 was released this morning.

**Comment by MikaelSmith** [comment]
Failed lint check, I'll get up maint PR.

**Comment by MikaelSmith** [comment]
Fixed in https://github.com/puppetlabs/pxp-agent/pull/627.

**Pull request merged**